cmake/configure/configure_2_trilinos.cmake

```diff
@@ -148,40 +161,124 @@ MACRO(FEATURE_TRILINOS_FIND_EXTERNAL var)
     # with the -std=c++0x flag of GCC, see deal.II FAQ.
     # Test whether that is indeed the case
     #
-    IF(NOT TRILINOS_SUPPORTS_CPP11)
-
-      IF(TRILINOS_HAS_C99_TR1_WORKAROUND)
-        LIST(APPEND TRILINOS_DEFINITIONS "HAS_C99_TR1_CMATH")
-        LIST(APPEND TRILINOS_USER_DEFINITIONS "HAS_C99_TR1_CMATH")
-      ELSE()
-        MESSAGE(STATUS "Could not find a sufficient Trilinos installation: "
-          "The installation is not compatible with the C++ standard selected for "
-          "this compiler."
+    IF(${DEAL_II_TRILINOS_WITH_SACADO})
+      #
+      # Look for Sacado_config.h - we'll query it to determine C++11 support:
+      #
+      DEAL_II_FIND_FILE(SACADO_CONFIG_H Sacado_config.h
+        HINTS ${Trilinos_INCLUDE_DIRS}
+        NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH
+        NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH
+        )
+      
+      SET(TRILINOS_WITH_MANDATORY_CXX11 FALSE)
+      IF(EXISTS ${SACADO_CONFIG_H})
+        #
+        # Determine whether Trilinos was configured with C++11 support and
+        # enabling C++11 in deal.II is mandatory.
+        #
+        FILE(STRINGS "${SACADO_CONFIG_H}" SACADO_CXX11_STRING
+          REGEX "#define HAVE_SACADO_CXX11")
+        IF(NOT "${SACADO_CXX11_STRING}" STREQUAL "")
+          SET(TRILINOS_WITH_MANDATORY_CXX11 TRUE)
+        ENDIF()
+      ENDIF()
+      
+      #
+      # Some versions of Sacado_cmath.hpp do things that aren't compatible
+      # with the -std=c++0x flag of GCC, see deal.II FAQ.
+      # Test whether that is indeed the case:
+      #
+      
```

Comment:
It looks like we don't put a free line between the comments and subsequent checks.